language/move-vm/types/src/values/values_impl.rs

```diff
@@ -2517,4 +2512,49 @@ pub mod prop {
             (Just(layout), value_strategy)
         })
     }
+
+    impl ValueImpl {
+        pub fn as_move_value(&self, ty: &FatType) -> MoveValue {
```

Comment:
is this needed?

Reply:
This is used for proptest. We generate random ValueImpl and map it to MoveValue, and do two rounds of serialization/deserialization test and make sure the bytes remains the same.